### PR TITLE
Add initial xk6 documentation

### LINKED ISSUE #169: Add xk6 documentation
As discussed internally, we should add a high-level page in the docs that explains [xk6](https://github.com/k6io/xk6), what it is, how to use it, etc., without going too much into details about developing an extension or showing any Go code for that matter.

This should include a large visible warning about the experimental stage of the project, no API stability guarantees, etc., and maybe a list of trusted extensions (the same ones mentioned in the upcoming article?) and link to https://github.com/topics/xk6.

### PR DESCRIPTION
**Pull request by imiric**
This is mostly a draft at this point but I'd like some feedback to know if it's on the right track, what's missing and how to structure things better (see the `TODO`s).

Closes #169

### CODE REVIEW

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 1]
```diff
@@ -1,4 +1,197 @@
...
+- JS extensions expose Go methods which can optionally receive a
+  [`context.Context`](https://golang.org/pkg/context/#Context) instance as the first argument.
+  This context is used throughout the k6 codebase and contains embedded objects
+  which can be extracted to access the
+  [`goja.Runtime`](https://pkg.go.dev/github.com/dop251/goja#Runtime) for a
+  particular VU, and to get more information about the test in progress.
+  Take a look at
+  [`common.GetRuntime()`](https://pkg.go.dev/go.k6.io/k6@v0.33.0/js/common#GetRuntime),
+  [`lib.GetState()`](https://pkg.go.dev/go.k6.io/k6@v0.33.0/lib#GetState) and
+  [`lib.GetExecutionState()`](https://pkg.go.dev/go.k6.io/k6@v0.33.0/lib#GetExecutionState).
```
Comment (MStoykov) [inline src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:178]: This might drastically change depending on the changes around https://github.com/k6io/k6/issues/1802, so maybe we should first decide on whether we will do any of the changes proposed there before we document stuff we might break

Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:178]: Good point, but that will change in v0.34.0, right? Since we have versioned docs now, I think it's fine to mention the current API here.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 2]
```diff
@@ -1,4 +1,197 @@
 ---
 title: 'k6 Extensions'
-redirect: 'https://k6.io/blog/extending-k6-with-xk6'
----
\ No newline at end of file
+---
+
+Traditionally, extending k6 with custom functionality that isn't available in the
```
Comment (ppcano) [inline src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:5]: In Docs, I'd remove the whole historical intro. 

Additionally, I would add a label about version support like in Scenarios.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 3]
```diff
@@ -1,4 +1,197 @@
...
+
+To address these issues and allow the community to more easily adapt k6 to fit their
+needs, [k6 v0.29.0](https://github.com/k6io/k6/releases/tag/v0.29.0) introduced
+the [xk6 framework](https://github.com/k6io/xk6) and the concept of k6 extensions.
+
+
+## What are k6 extensions?
+
+k6 extensions are standalone Go projects that call k6 APIs but are otherwise
+unrestricted in their functionality. This provides the freedom for extension authors
```
Comment (ppcano) [inline src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:33]: In the beginning, we should focus on describing the benefits of this feature. 

Having the ability to use extensions to experiment with features that can become part of the feature is a "secondary" benefit, imho.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 4]
```diff
@@ -1,4 +1,197 @@
...
+  if the extension is inefficient.
+  As such you should ensure that `AddMetricSamples()` doesn't block for a long time,
+  and that metrics are flushed periodically to avoid memory leaks.
+
+  Since this is common functionality that most outputs should have, we've provided
+  a couple of helper structs: [`output.SampleBuffer`](https://pkg.go.dev/go.k6.io/k6@v0.33.0/output#SampleBuffer)
+  and [`output.PeriodicFlusher`](https://pkg.go.dev/go.k6.io/k6@v0.33.0/output#PeriodicFlusher)
+  that output implementations can use which handles bufferring and periodic
+  flushing in a thread-safe way.
+  For usage examples see the [`statsd` output](https://github.com/k6io/k6/blob/v0.33.0/output/statsd/output.go#L55).
```
Comment (ppcano) [inline src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:193]: Can you add an example of an extension instead of the k6 codebase?

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 5]
```diff
@@ -1,4 +1,197 @@
...
+A good starting point for using xk6 and writing your own extension is the [xk6
+introductory article](https://k6.io/blog/extending-k6-with-xk6).
+<!-- TODO: How much of this article do we want to reproduce here? -->
+
+You'll first need to decide the type of extension you wish to create. Choose a
+JavaScript extension if you want to extend the JS functionality of your script, or an
+Output extension if you want to process the metrics emitted by k6 in some way.
+
+A few things to keep in mind when writing a new extension:
+<!-- TODO: Better structure this list? -->
```
Comment (ppcano) [inline src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:158]: Agree. It is a bit hard to follow. 

I wonder if. it could be good to show snippets of relevant extension code to distinguish between the various sections and make the whole section more pleasant to read.

**Comment by ppcano** [comment]
It's a good start @imiric. I have added a few comments about the first impressions, take the feedback as suggestions or to think about.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 2 above)
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:5]: Hhmm I think explaining the issues extensions were meant to solve serves as a good introduction for why someone might want to use them. Starting from "What are k6 extensions?" would be too abrupt IMO. Unless you have another suggestion for the intro :)

And sure, I'll add a more prominent version support notice.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 3 above)
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:33]: This is mostly a definition paragraph, but what other benefits would you mention?

The historical introduction explains the issues extensions are meant to solve for end users, so I'm not sure what else to mention here.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 4 above)
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:193]: Hhmm we already have a lot of external examples and I thought it would be good to link to an internal module for once, to showcase that extensions and internal modules are near identical in implementation.

Besides, I'm not sure there are any extensions that actually use this yet, since it hasn't been documented ;)

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 5 above)
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:158]: Yeah, could be... In #169 we kind of agreed to not show any Go code here, but we could split this section into "Writing a new JS extension" (linking to the existing article) and "Writing a new Output extension", and do a small walkthrough with some snippets of code instead of a complete extension like in the article.

I'll think about it, and more opinions are welcome. :)

**Comment by simskij** [comment]
What's the status here, @imiric and @ppcano? Is this abandoned, finished, or still ongoing?

**Comment by imiric** [comment]
@simskij It's in progress and awaiting more feedback. I actually picked it back up today to address [this](https://github.com/grafana/k6-docs/pull/346#discussion_r662003317) and [this](https://github.com/grafana/k6-docs/issues/169#issuecomment-860766221), but new comments are welcome.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 5 above)
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md]: @ppcano Take a look at the latest changes. It kind of breaks the no-code rule, but I really don't think we can document writing an extension without showing code... :)

**Comment by imiric** [comment]
I expanded more on the writing extension part, but it will need fact checking from @na-- and @MStoykov :sweat_smile:

It's still a bit WIP and there are some `TODO`s and ellipsis that should be addressed, but there shouldn't be much left after that.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 6]
```diff
@@ -1,4 +1,393 @@
...
+  is a thread-safe buffer for metric samples to help with memory management and
+  [`output.PeriodicFlusher`](https://github.com/grafana/k6/blob/v0.33.0/output/helpers.go#L75)
+  will periodically run a function which is useful for flushing or dispatching the
+  buffered samples.
+  For usage examples see the [`statsd` output](https://github.com/k6io/k6/blob/v0.33.0/output/statsd/output.go#L55).
+
+
+#### Things to keep in mind
+
+- ...
```
Comment (imiric) [inline src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:393]: Any ideas what to mention here? I considered memory management and no blocking, but it's mentioned above.

Should we drop or rename this section?

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 7]
```diff
@@ -1,4 +1,395 @@
...
+  scenario this output might have to be buffered and flushed periodically to avoid
+  memory leaks. Below we'll discuss some helpers you can use for that.
+
+
+#### Additional features
+
+- Output structs can optionally implement additional interfaces that allows them to
+  [receive thresholds](https://github.com/grafana/k6/blob/v0.33.0/output/types.go#L79),
+  [test run status updates](https://github.com/grafana/k6/blob/v0.33.0/output/types.go#L94)
+  or [interrupt a test run](https://github.com/grafana/k6/blob/v0.33.0/output/types.go#L88).
```
Comment (imiric) [inline src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:381]: Should these link to pkg.go.dev or GitHub? Our API docs aren't great, so I thought GH would be better.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 6 above)
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md]: Couldn't think of anything, so decided to drop it.

**Comment by imiric** [comment]
This is ready for review again. @na-- @MStoykov @ppcano :pray:

@simskij I mentioned the supported modes of execution in 56b3702, but can you clarify what you meant by "considerations and drawbacks"? I'm not sure what else to mention besides whether custom k6 binaries are supported or not. Suggestions welcome :)

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 8]
```diff
@@ -1,4 +1,400 @@
...
+## Supported modes of execution
+
+xk6 and the examples above work great for local execution, i.e. running a
+script with a binary built by xk6 on your own infrastructure. Custom k6 binaries are
+also supported by the [k6 Kubernetes operator](https://github.com/grafana/k6-operator#using-extensions).
+
+They're currently not supported in [k6 Cloud](https://k6.io/docs/cloud/), but adding support
+is on our roadmap. However, it is possible to run a binary built with xk6 and send
+metrics produced by extension code to the Cloud using the
+[`cloud` output](https://k6.io/docs/results-visualization/cloud/).
```
Comment (imiric) [inline src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:400]: Should we add some kind of outro here, relevant links, etc.?

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 9]
```diff
@@ -1,4 +1,400 @@
...
+
+## Writing a new extension
+
+The first thing you should do before starting work on a new extension is to confirm
+that a similar extension doesn't already exist for your use case. Take a look at
+the [Ecosystem page](/ecosystem) and the [`xk6` topic on GitHub](https://github.com/topics/xk6).
+For example, if a system you need support for can be tested with a generic protocol
+like MQTT, prefer using [xk6-mqtt](https://github.com/pmalhaire/xk6-mqtt)
+instead of creating an extension that uses some custom protocol.
+
```
Comment (MStoykov) [inline src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:163]: I would also argue to try to implement the functionality in pure JS, which will in general will be way more supported and likely easier to write and reuse

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 10]
```diff
@@ -1,4 +1,400 @@
...
+  dependencies required by a specific extension.
+- Simple API for Go programmers that handles the Go<->JS translation, with the
+  ability to call any public k6 Go API. Extensions are first-class components along
+  with other built-in modules.
+- Cross-platform like Go and runs great on macOS, Windows and Linux.
+
+
+### Extension types
+
+The initial version of xk6 released in v0.29.0 supported only JavaScript extensions,
```
Comment (ppcano) [inline src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:61]: imo, it is unnecessary to mention again the 0.29 / 0.31 versions

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 11]
```diff
@@ -1,4 +1,400 @@
...
+
+- By forking the [k6 repository](https://github.com/k6io/k6), doing the changes in Go
+  and submitting a pull request for review by the core team (preferably following
+  the [contribution guidelines](https://github.com/k6io/k6/blob/master/CONTRIBUTING.md)).
+  This is a great way to contribute new core features, but it can be a lengthy
+  process, and submissions might be rejected if they don't align with the long-term
+  vision of the project.
+
+To address these issues and allow the community to more easily adapt k6 to fit their
+needs, [k6 v0.29.0](https://github.com/k6io/k6/releases/tag/v0.29.0) introduced
```
Comment (ppcano) [inline src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:28]: imo, it is unnecessary to mention again the 0.29 / 0.31 versions

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 3 above)
Comment (ppcano) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:35]: Also about https://github.com/grafana/k6-docs/pull/346/files#r661994675

My view here is that the intro and the `what is k6 extension` and `what is xk6` sections are a little verbose.  For the Intro sections, I rather going more direct to the point in Docs and extend the historical in the blog post.

I like `Extension Types` section, but perhaps the `Getting Started` section could be better connected with the intro.

The article is good. If you want, I can give it a try in the future for a new proposal of the intro.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 2 above)
Comment (ppcano) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:7]: https://github.com/grafana/k6-docs/pull/346/files#r678100601

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 12]
```diff
@@ -1,4 +1,400 @@
...
+  which is a bit more idiomatic to JS.
+
+- Methods that specify `context.Context` or `*context.Context` as the first
+  argument will be passed the `Context` instance used internally in k6,
+  which has attached some useful objects for inspecting the internal execution
+  state, such as
+  [`lib.State`](https://github.com/grafana/k6/blob/v0.33.0/lib/state.go#L43)
+  or VU state, [`lib.ExecutionState`](https://github.com/grafana/k6/blob/v0.33.0/lib/execution.go#L142),
+  and the [`goja.Runtime`](https://github.com/dop251/goja/blob/705acef95ba3654f89c969d9e792ac5f49215350/runtime.go#L162) instance
+  the VU is using to execute the script.
```
Comment (codebien) [inline src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:275]:
```suggestion
  and the [`goja.Runtime`](https://github.com/dop251/goja/blob/705acef95ba3654f89c969d9e792ac5f49215350/runtime.go#L162) instance ?
  the VU is using to execute the script.
```
Seems the period is not completed :thinking:

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 13]
```diff
@@ -1,4 +1,400 @@
...
+The k6 Go-JS bridge has a few features we should highlight:
+
+- Go method names will be converted from Pascal case to Camel case when
+  accessed in JS, as in the example above: `IsGreater` becomes `isGreater`.
+
+<!-- FIXME: Markdown magic to escape backticks in Go struct tags? <pre> doesn't work... -->
+- Similarly, Go field names will be converted from Pascal case to Snake case.
+  For example, the struct field `SomeField string` is accessible in JS
+  as the `some_field` object property. This behavior is configurable with the
+  `js` struct tag, so this can be changed with `SomeField string ``js:"someField"```
```
Comment (codebien) [inline src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:248]: the backtick escape is not working on staging

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 14]
```diff
@@ -1,4 +1,400 @@
...
+
+#### Things to keep in mind
+
+- The code in the `default` function (or another function specified by
+  [`exec`](/using-k6/scenarios/#common-options)) will be executed many
+  times during a test run and possibly in parallel by thousands of VUs.
+  As such any operation of your extension meant to run in that context
+  needs to be performant and [thread-safe](https://en.wikipedia.org/wiki/Thread_safety).
+- Any heavy initialization should be done in the [`init`
+  context](/javascript-api/init-context/)...
```
Comment (codebien) [inline src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:290]:
```suggestion
- Any heavy initialization should be done in the [init
  context](/javascript-api/init-context/).
```

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 15]
```diff
@@ -1,4 +1,400 @@
...
+- Custom metric emission can be done by creating new metrics using
+  [`stats.New()`](https://github.com/grafana/k6/blob/v0.33.0/stats/stats.go#L449)
```
Comment (codebien) [inline src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:292]:
```suggestion
  [stats.New()](https://github.com/grafana/k6/blob/v0.33.0/stats/stats.go#L449)
```

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 7 above)
Comment (codebien) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:379]: I'm fine with GitHub, should we use permalinks?

**Review by codebien — CHANGES_REQUESTED** [review]
LGTM, just some minor comments

As a general comment, I would avoid using the backticks with links (e.g. [`foo` bar]) because they overlay/mix the rendered format for the links.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 16]
```diff
@@ -1,4 +1,400 @@
...
+#### Output extension
+
+While k6 has built-in support for many popular [output
+backends](/docs/getting-started/results-output/), this list will undoubtedly not be
+exhaustive. Support for new systems and novel ways of handling the metric data
+generated by k6 can be easily added with Output extensions. These receive metric
+samples from k6, and are able to do any processing or further dispatching.
+
+Some examples include: [xk6-prometheus](https://github.com/szkiba/xk6-prometheus)
+and [xk6-influxdbv2](https://github.com/li-zhixin/xk6-influxdbv2).
```
[the remark this answers is not in the document]
Comment (MStoykov) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:86]: yeah probably better, but also maybe at least one that isn't our own extension, so the prometheus one seems fine

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 7 above)
Comment (MStoykov) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:379]: I am for pkg.go.dev. This links will get stale really fast. It's better to link to the docs and improve that over time than to just link to code. Also the docs links to the code so people can use that.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 17]
```diff
@@ -1,4 +1,400 @@
...
+good fit if you want to extend the JS functionality of your script, or add support
+for a new network protocol to test with. An Output extension would be more suitable
+if you need to process the metrics emitted by k6 in some way, submit them to a
+specific storage backend that was previously unsupported, etc. The k6 APIs you'll
+need to use and things to consider while developing will be different in each case.
+
+
+### Writing a new JavaScript extension
+
+A good starting point for using xk6 and writing a JS extension is the [xk6
```
Comment (MStoykov) [inline src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:174]: Given that most users will have problems with the context passing after some point I do think that this section should either have a disclaimer that this is an extremely simple example or to be expanded to show the "module v2" way and possibly a note about common.Bind and ... not using it ;)

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 18]
```diff
@@ -1,4 +1,400 @@
...
+
+</CodeGroup>
+
+This will build a `k6` binary in the current directory based on k6 v0.33.0, bundling
+a JavaScript and an Output extension. Now you can run a script with this binary that
+uses the [`xk6-sql` JS API](https://github.com/imiric/xk6-sql) and the
+[Prometheus output](https://github.com/szkiba/xk6-prometheus).
+
+Note that when running the script we have to specify the binary just built in the
+current directory (`./k6`), as otherwise some other `k6` binary found on the system
```
[the remark this answers is not in the document]
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:149]: I think they're different usages: k6/xk6 when talking about the project itself, and `k6`/`xk6` when talking about the command, CLI, etc. So in this case I think `k6` makes more sense.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 12 above)
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:275]: Not sure what you mean, it looks fine on https://mdr-ci.staging.k6.io/docs/refs/pull/346/merge/misc/k6-extensions/.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 7 above)
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md]: Changed links to `pkg.go.dev` in 51e583a.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 9 above)
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md]: Sure, done in 3316dc2.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 10 above)
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md]: OK, done in 0ec0f65.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 11 above)
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md]: OK, done in 0ec0f65.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 19]
```diff
@@ -1,4 +1,400 @@
...
+`$GOPATH/bin` is added to your `$PATH` environment variable. See the
+[Go documentation](https://golang.org/cmd/go/#hdr-GOPATH_environment_variable) for details.
+
+Once xk6 is installed, building a k6 binary with one or more extensions can be done
+with the following command:
+
+<CodeGroup labels={[]} lineNumbers={[false]}>
+
+```bash
+$ xk6 build v0.33.0 \
```
[the remark this answers is not in the document]
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md]: Sure, done in ecc287f.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md [hunk 20]
```diff
@@ -1,4 +1,400 @@
...
+  --with github.com/imiric/xk6-sql \
+  --with github.com/szkiba/xk6-prometheus
+```
+
+</CodeGroup>
+
+This will build a `k6` binary in the current directory based on k6 v0.33.0, bundling
+a JavaScript and an Output extension. Now you can run a script with this binary that
+uses the [`xk6-sql` JS API](https://github.com/imiric/xk6-sql) and the
```
[the remark this answers is not in the document]
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md]: OK, done in ecc287f.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md]: OK, changed in ecc287f.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 14 above)
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md]: Changed in ecc287f.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 13 above)
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md]: Yeah :disappointed: I changed it to `<code>SomeField string &grave;js:"someField"&grave;</code>` in ecc287f, hopefully it renders fine :man_shrugging:

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (codebien) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:149]: it makes sense, it wasn't clear for me but I haven't suggestions for improving it. So it's fine for me.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 15 above)
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md]: I see your point that links on preformatted text are invisible without hovering over them, but that's a UI problem, not a semantic one :)

So I'd rather leave this and other usage as is, and create an issue for someone from frontend (PixelPoint?) to take a look at fixing the rendering. @ppcano WDYT?

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 17 above)
Comment (imiric) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:177]: Hhmm yeah, that's tricky. On the one hand, it's going to be confusing if we have to document the `common.Bind()` way and the `ModuleV2` way... But on the other, suggesting `common.Bind()` would be outdated quickly and we probably don't want to suggest that approach moving forward, and I'm not sure if `ModuleV2` can be considered stable for public use yet.

@na-- WDYT?

Comment (MStoykov) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md:177]: I am for just documenting `ModuleV2` with a warning (that should be either way there) that internal APIs can change and that the whole thing is alpha like. We can also mention `common.Bind`, link to the issue I made about it and say that it's *not* advisable to use it at all and moving to `ModuleV2` is the better plan. 

I doubt we will just remove `common.Bind` in 2-6 months and likely `ModuleV2` will just get a lot more methods. The only one (unfortunately) that I don't like is `GetRuntime` and mostly due to the whole fact that it lets you have the Runtime indefinitely even if (in the future) we have something asynchronous going on as well. But at least I don't have any non-trivial and likely non-performant ideas for it, so we likely will have to keep it and have a long explanation on how ... you might want to be careful with it, one day.

File: src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md (hunk 13 above)
Comment (codebien) [reply src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md]: Screenshot from 2021-10-06 18-55-26
I see this

**Comment by ppcano** [comment]
@na-- @MStoykov Any blocker here? I suggest merging it and iterating it next.  I want to migrate this content to the [Extensions section](https://k6.io/docs/extensions/) and will require some minor changes.

**Comment by imiric** [comment]
@codebien It doesn't and good idea. I mentioned it in db1c351.
